nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -9,7 +9,10 @@ let
 
   ipfsFlags = ''${if cfg.autoMigrate then "--migrate" else ""} ${if cfg.enableGC then "--enable-gc" else ""} ${toString cfg.extraFlags}'';
 
-  pathEnv = { IPFS_PATH = cfg.dataDir; };
+  # Before Version 17.09, ipfs would always use "/var/lib/ipfs/.ipfs" as it's dataDir
+  defaultDataDir = if versionAtLeast config.system.stateVersion "17.09" then
+    "/var/lib/ipfs" else
+    "/var/lib/ipfs/.ipfs";
 
```

Comment:
I keep thinking about this and I just can't come up with a nice solution, maybe just put it in the release notes?